### PR TITLE
add MobileNetV2 model

### PR DESCRIPTION
**Pull request by hollance**
# What does this PR do?

Adds MobileNetV2 to the Transformers library.

This includes an image classification head and a basic DeepLabV3+ semantic segmentation head.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Very clean, thanks a lot for the new model!

File: tests/models/mobilenet_v2/test_modeling_mobilenet_v2.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+                self.image_size // self.output_stride,
+            ),
+        )
+        self.parent.assertEqual(
+            result.pooler_output.shape,
+            (
+                self.batch_size,
+                self.last_hidden_size,
+            ),
+        )
```
[the remark this answers is not in the document]
Comment (hollance) [reply tests/models/mobilenet_v2/test_modeling_mobilenet_v2.py]: OK, done.

File: src/transformers/models/mobilenet_v2/image_processing_mobilenet_v2.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    PILImageResampling,
+    is_batched,
+    to_numpy_array,
+    valid_images,
+)
+from ...utils import logging
+
+
+if is_vision_available():
+    import PIL
```
Comment (sgugger) [inline src/transformers/models/mobilenet_v2/image_processing_mobilenet_v2.py:47]: Apparently this is not used according to `flake8`.

File: src/transformers/models/mobilenet_v2/image_processing_mobilenet_v2.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    def post_process_semantic_segmentation(self, outputs, target_sizes: List[Tuple] = None):
+        """
+        Args:
+        Converts the output of [`MobileNetV2ForSemanticSegmentation`] into semantic segmentation maps. Only supports
+        PyTorch.
+            outputs ([`MobileNetV2ForSemanticSegmentation`]):
+                Raw outputs of the model.
+            target_sizes (`List[Tuple]`, *optional*):
+                A list of length `batch_size`, where each item is a `Tuple[int, int]` corresponding to the requested
+                final size (height, width) of each prediction. If left to None, predictions will not be resized.
```
Comment (sgugger) [inline src/transformers/models/mobilenet_v2/image_processing_mobilenet_v2.py:349]: Args should be below the docstring.

**Review by sgugger — COMMENTED** [review]
A couple of final nits before we merge this :-)

File: src/transformers/models/mobilenet_v2/image_processing_mobilenet_v2.py (hunk 2 above)
Comment (hollance) [reply src/transformers/models/mobilenet_v2/image_processing_mobilenet_v2.py]: yeah, I was too enthusiastic copy-pasting stuff :-)

**Comment by sgugger** [comment]
Good to merge @hollance ?

**Comment by hollance** [comment]
@sgugger It seems to fail some tests but they don't look they're from my changes? If you're OK with that test failing then I think this is ready to get merged.

**Comment by sgugger** [comment]
Failure is a flaky test, unrelated to this PR.

**Pull request merged**